File: src (cpp)/animal/mammalia/zebra.cpp
#include "zebra.h"

Zebra::Zebra(int _x, int _y, int _weight) : def_ID(4),
                                            def_ratio_meat(0),
                                            def_ratio_plant(45) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Zebra::Interact() {
  std::cout << "The zebra is running at the grass field" << std::endl;
}

File: src/virtualzoo/animal/mammalia/Zebra.java
package virtualzoo.animal.mammalia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Herbivora;

/**
 * Kelas Zebra mendefinisikan atribut untuk zebra.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Zebra extends Mammalia {

  /** <p>
   * Constructor
   *
   * Menciptakan zebra dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat zebra.
   */
  public Zebra(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Herbivora(animalWeight, 45);
    id = 4;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan zera.
   */
  public void interact() {
    System.out.println("The zebra is running at the grass field");
  }
}
